### ISSUE TITLE
Misleading diagnostic when trying to use Box<Error> for io::Error:new

### ISSUE THREAD

**Issue by vi** [rust-lang/rust#50489]
```rust
fn main(){
    let q : Box<::std::error::Error /* + Send + Sync */ > = unimplemented!();
    let w : ::std::io::Error = ::std::io::Error::new(::std::io::ErrorKind::Other, q);
}
```

First compiler error message:

```
error[E0277]: the trait bound `std::error::Error: std::marker::Sized` is not satisfied
 --> src/main.rs:3:32
  |
3 |     let w : ::std::io::Error = ::std::io::Error::new(::std::io::ErrorKind::Other, q);
  |                                ^^^^^^^^^^^^^^^^^^^^^ `std::error::Error` does not have a constant size known at compile-time
  |
  = help: the trait `std::marker::Sized` is not implemented for `std::error::Error`
  = note: required because of the requirements on the impl of `std::error::Error` for `std::boxed::Box<std::error::Error>`
  = note: required because of the requirements on the impl of `std::convert::From<std::boxed::Box<std::error::Error>>` for `std::boxed::Box<std::error::Error + std::marker::Send + std::marker::Sync + 'static>`
  = note: required because of the requirements on the impl of `std::convert::Into<std::boxed::Box<std::error::Error + std::marker::Send + std::marker::Sync + 'static>>` for `std::boxed::Box<std::error::Error>`
  = note: required by `std::io::Error::new`
```

Later it gets to the `Send` and `Sync` parts, but initially it makes one wonder about why wanting `Sized` to `Error` itself instead of `Box<Error>`.

**Comment by estebank** [comment]
Triage: only change is [pointing at the correct argument](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=6250489e42fbc42a77945b59b8b73991).
